requester/requester.go

```diff
@@ -111,27 +111,32 @@ func (b *Work) makeRequest(c *http.Client) {
 	var dnsStart, connStart, resStart, reqStart, delayStart time.Time
 	var dnsDuration, connDuration, resDuration, reqDuration, delayDuration time.Duration
 	req := cloneRequest(b.Request, b.RequestBody)
+	redirects := 0
 	if b.EnableTrace {
 		trace := &httptrace.ClientTrace{
 			DNSStart: func(info httptrace.DNSStartInfo) {
 				dnsStart = time.Now()
 			},
 			DNSDone: func(dnsInfo httptrace.DNSDoneInfo) {
-				dnsDuration = time.Now().Sub(dnsStart)
+				dnsDuration += time.Now().Sub(dnsStart)
 			},
 			GetConn: func(h string) {
+				redirects++
+				if redirects > 1 {
```

Comment:
Don't we need a custom RoundTripper to properly measure this?

```
type traceTransport struct{
    current *http.Request
}

func (t *traceTransport) Current() *http.Request {
    return t.current
}

func (rt *traceTransport) RoundTrip(req *http.Request) (*http.Response, error) {
        t.current = req
        return http.DefaultTransport.RoundTrip(req)
}

latencies := make(map[*http.Request]struct{
     dnsStart, connStart, resStart, reqStart, delayStart time.Time
     dnsDuration, connDuration, resDuration, reqDuration, delayDuration time.Duration
})

tr := &traceTransport{}
client := &http.Client{Transport: tr}

traceClient := &httptrace.ClientTrace{
            GotConn: func(info httptrace.GotConnInfo) {
                        // use tr.Current() to modify latencies.
            }
}
```

Is it an overkill?